plugins/catalog-backend/src/ingestion/HigherOrderOperations.ts

```diff
@@ -94,16 +94,24 @@ export class HigherOrderOperations implements HigherOrderOperation {
     if (!previousLocation) {
       await this.locationsCatalog.addLocation(location);
     }
-    const outputEntities: Entity[] = [];
-    for (const entity of readerOutput.entities) {
-      const out = await this.entitiesCatalog.addOrUpdateEntity(
-        entity.entity,
-        location.id,
-      );
-      outputEntities.push(out);
+    if (readerOutput.entities.length === 0) {
+      return { location, entities: [] };
     }
 
-    return { location, entities: outputEntities };
+    const writtenEntities = await this.entitiesCatalog.batchAddOrUpdateEntities(
+      readerOutput.entities,
+      location.id,
+    );
+
+    if (writtenEntities.length === 0) {
+      return { location, entities: [] };
+    }
+
+    const entities = await this.entitiesCatalog.entities({
+      'metadata.uid': writtenEntities.map(e => e.entityId),
```

Comment:
i suspect that this breaks for a large write of a hundred thousand items (i sometimes use the ldap import as a stress test and it makes things explode a lot that one assumed was working fine)

Reply:
Yep, fixing it in the DB layer requires hassle of temporary tables though. I suspect the best workaround might be to batch the writes here.

Reply:
Mm, maybe that's better to get the width under explicit control.